src/main/java/core/basesyntax/ColorSupplier.java

```diff
@@ -1,7 +1,12 @@
 package core.basesyntax;
 
-public class ColorSupplier {
+import java.util.Random;
+
+public class ColorSupplier extends Ball{
     public String getRandomColor() {
-        return null;
+        int index = new Random().nextInt(Colors.values().length);
+        Colors colors = Colors.values()[index];
+        setColor(colors.toString());
```

Comment:
The setColor method call is not necessary here and is also incorrect because ColorSupplier should not have a state related to a specific Ball. You should only return the color as a string.